tests/e2e/slashing.go

```diff
@@ -374,106 +375,396 @@ func (suite *CCVTestSuite) TestOnRecvSlashPacketErrors() {
+		powerBeforeUndelegate = validator.GetConsensusPower(sdk.DefaultPowerReduction)
+		expectedPowerDelegated := (bondAmt.Quo(sdk.DefaultPowerReduction)).Int64()
+		suite.Require().Equal(
+			powerBeforeDelegate+expectedPowerDelegated,
+			powerBeforeUndelegate,
+			"unexpected power after delegation; test: "+tc.name,
+		)
 
-	// save unbonding balance before slashing tests
-	ubd, found := providerStakingKeeper.GetUnbondingDelegation(
-		suite.providerChain.GetContext(), delAddr, valAddr)
-	suite.Require().True(found)
-	ubdBalance := ubd.Entries[0].Balance
+		// undelegate half of the delegated shares
+		vscID = undelegate(suite, delAddr, valAddr, shares.QuoInt64(2))
+		// - check that staking unbonding op was created and onHold is true
+		checkStakingUnbondingOps(suite, 1, true, true, "test: "+tc.name)
+		// - check that CCV unbonding op was created
+		checkCCVUnbondingOp(suite, suite.providerCtx(), suite.consumerChain.ChainID, vscID, true, "test: "+tc.name)
+		// - check undelegation entry balance
+		ubd, found := providerStakingKeeper.GetUnbondingDelegation(suite.providerCtx(), delAddr, valAddr)
+		suite.Require().True(found)
+		suite.Require().True(ubd.Entries[0].Balance.Equal(bondAmt.Quo(sdk.NewInt(2))))
+
+		// commit state on provider
+		suite.coordinator.CommitBlock(suite.providerChain)
+
+		// get the power after undelegate
+		validator = suite.getVal(suite.providerCtx(), valAddr)
+		powerAfterUndelegate = validator.GetConsensusPower(sdk.DefaultPowerReduction)
+		expectedPowerDelegated = (bondAmt.Quo(sdk.DefaultPowerReduction).Quo(sdk.NewInt(2))).Int64()
+		suite.Require().Equal(
+			powerBeforeDelegate+expectedPowerDelegated,
+			powerAfterUndelegate,
+			"unexpected power after undelegation; test: "+tc.name,
+		)
 
-	for _, tc := range testCases {
-		slashPacket := ccv.NewSlashPacketData(
-			abci.Validator{Address: tmValidator.Address, Power: tmValidator.VotingPower},
-			tc.vscID,
-			stakingtypes.Downtime,
+		// relay VSCPacket w/ undelegation from provider to consumer
+		relayAllCommittedPackets(
+			suite,
+			suite.providerChain,
+			suite.path,
+			ccv.ProviderPortID,
+			suite.path.EndpointB.ChannelID,
+			1,
+			"test: "+tc.name,
 		)
 
-		// slash
-		providerKeeper.HandleSlashPacket(suite.providerChain.GetContext(), suite.consumerChain.ChainID, *slashPacket)
+		// get the height when the consumer "applied" the undelegation
+		maturityTimes = consumerKeeper.GetAllPacketMaturityTimes(suite.consumerCtx())
+		suite.Require().Len(maturityTimes, 2, "unexpected number of maturity times; test: "+tc.name)
+		vscID = maturityTimes[1].VscId
+		hToVSCids = consumerKeeper.GetAllHeightToValsetUpdateIDs(suite.consumerCtx())
+		found = false
+		for _, hToVSCid := range hToVSCids {
+			if hToVSCid.ValsetUpdateId == vscID {
+				undelegateConsumerHeight = hToVSCid.Height
+				found = true
+				break
+			}
+		}
+		suite.Require().True(found, "cannot find height mapped to vscID; test: "+tc.name)
```

Comment:
Maybe we can move this into a helper function that takes a VSC id as input and either returns a height or fails.